modules/abstract-eth/src/abstractEthLikeNewCoins.ts

```diff
@@ -2725,6 +2766,178 @@ export abstract class AbstractEthLikeNewCoins extends AbstractEthLikeCoin {
     return {};
   }
 
+  /**
+   * Get forwarder factory and implementation addresses for deposit address verification.
+   * Forwarders are smart contracts that forward funds to the base wallet address.
+   *
+   * @param {number | undefined} walletVersion - The wallet version
+   * @returns {object} Factory and implementation addresses for forwarders
+   */
+  getForwarderFactoryAddressesAndForwarderImplementationAddress(walletVersion: number | undefined): {
+    forwarderFactoryAddress: string;
+    forwarderImplementationAddress: string;
+  } {
+    const ethNetwork = this.getNetwork();
+
+    switch (walletVersion) {
```

Comment:
what about walletVersion 1? and I would suggest to take forwarderVersion as a param instead of walletVersion

Reply:
just saw that default branch is taking care of walletVersion 1, better keep it explicit case instead of default, that looks more readable to me